pkg/lightning/restore/restore.go

```diff
@@ -2733,3 +2933,789 @@ func (cr *chunkRestore) restore(
 		return ctx.Err()
 	}
 }
+
+type metaMgrBuilder interface {
```

Comment:
let's move these meta-manager stuff into its own `.go` file to avoid further bloating the `restore.go` file (it's over 3700 lines after this PR 😅)